### PR TITLE
#161966897 Implement tests for Articles module

### PR DESCRIPTION
**Pull request by PromasterGuru**
#### What does this PR do?
Implement tests for Articles module models

#### Description of Task to be completed?
- Create failing tests for the Articles models
- Implement the models to make the test pass
- Eliminate unnecessary models

#### How should this be manually tested?
- Clone the App to your local machine: `https://github.com/andela/ah-infinity-stones.git`
- Cd to the project directory: `cd ah-infinity-stones`
- Create a virtual environment: `virtualenv -p python3 venv`
- Activate the virtual environment: `source venv/bin/activate`
- Install requirements: `pip3 install -r requirements.txt`

- Create a .env file with all the database configuration variables. See the ReadMe file.
- Export your database configuration variables: `source .env`
- Make migrations: `python3 manage.py makemigrations && python3 manage.py migrate`
- Run tests: `python3 manage.py test`

#### What are the relevant pivotal tracker stories?
[#161966897](https://www.pivotaltracker.com/story/show/161966897)

#### Screenshots (if appropriate)
screenshot 2018-12-05 at 19 42 27

### CODE REVIEW

**Comment by ThaDeveloper** [comment]
Great stuff all the model tests are passing. Include how to manually test the PR

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,6 @@ __pycache__/
 *.py[cod]
 *$py.class
 .vscode/
-.env
```
Comment (r-wambui) [inline .gitignore:6]: Why did you decide to have `.env` removed from the gitignore file

File: authors/apps/articles/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        self.sections.save()
+        last_count = Highlight.objects.count()
+        self.assertNotEqual(first_count, last_count)
+
+    def test_comment_history_model_can_be_created(self):
+        """Test the model can create Comment History table"""
+        first_count = CommentHistory.objects.count()
+        self.history.save()
+        last_count = CommentHistory.objects.count()
+        self.assertNotEqual(first_count, last_count)
```
Comment (r-wambui) [inline authors/apps/articles/tests/test_models.py:123]: Keep up the good use of pep 8 on this file.

File: authors/apps/articles/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class CreateArticleTestCase(TestCase):
+    """This class defines the test suit for the Article model"""
+    pass
+
+
+class ViewArticleTestCase(TestCase):
+    """This class defines the test suit for api views"""
+    pass
```
Comment (r-wambui) [inline authors/apps/articles/tests/test_views.py:15]: The previous PR was testing the views. Why do we still have this?

**Review by amutava — COMMENTED** [review]
Hi, I can see you have tests for the models. Did you forget to include tests for the api. Kindly advise.

File: .gitignore (hunk 1 above)
Comment (PromasterGuru) [reply .gitignore:6]: Hey, by default the file had already been ignored hence that was just a repetition.

**Comment by PromasterGuru** [comment]
@andela-amutava We were paired with @kamyaD. We are working on the test for articles and comments on his machine and on different branches.

File: authors/apps/articles/tests/test_views.py (hunk 3 above)
Comment (r-wambui) [inline authors/apps/articles/tests/test_views.py:15]: You had a previous PR with test on views. Maybe you should have this reflect in that PR instead

Comment (PromasterGuru) [reply authors/apps/articles/tests/test_views.py:15]: Hey @r-wambui,
My previous PR was testing Article module models, we are working on the views on a different branch with @kamyaD though I haven't rebased the new branch.

Comment (PromasterGuru) [reply authors/apps/articles/tests/test_views.py:15]: Let me check on that.

Comment (PromasterGuru) [reply authors/apps/articles/tests/test_views.py:15]: That seems to had been reflected on @kamyaD PR. We are working it.

**Comment by leonardnjura** [comment]
The article models will be so handy as we implement the features moving forward. Cool stuff @PromasterGuru

**Pull request merged**